src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenAsyncTests.vb

```diff
@@ -9251,6 +9243,2032 @@ End Module"
+    IL_0033:  ldarg.0
+    IL_0034:  ldflda     ""Program.VB$StateMachine_0_Assign.$Builder As System.Runtime.CompilerServices.AsyncTaskMethodBuilder""
+    IL_0039:  ldloca.s   V_1
+    IL_003b:  ldarg.0
+    IL_003c:  call       ""Sub System.Runtime.CompilerServices.AsyncTaskMethodBuilder.AwaitUnsafeOnCompleted(Of System.Runtime.CompilerServices.TaskAwaiter(Of Integer), Program.VB$StateMachine_0_Assign)(ByRef System.Runtime.CompilerServices.TaskAwaiter(Of Integer), ByRef Program.VB$StateMachine_0_Assign)""
+    IL_0041:  leave.s    IL_00b1
+    IL_0043:  ldarg.0
+    IL_0044:  ldc.i4.m1
+    IL_0045:  dup
+    IL_0046:  stloc.0
+    IL_0047:  stfld      ""Program.VB$StateMachine_0_Assign.$State As Integer""
+    IL_004c:  ldarg.0
+    IL_004d:  ldfld      ""Program.VB$StateMachine_0_Assign.$A0 As System.Runtime.CompilerServices.TaskAwaiter(Of Integer)""
+    IL_0052:  stloc.1
+    IL_0053:  ldarg.0
+    IL_0054:  ldflda     ""Program.VB$StateMachine_0_Assign.$A0 As System.Runtime.CompilerServices.TaskAwaiter(Of Integer)""
+    IL_0059:  initobj    ""System.Runtime.CompilerServices.TaskAwaiter(Of Integer)""
+    IL_005f:  ldsflda    ""A.b As B""
+    IL_0064:  ldloca.s   V_1
+    IL_0066:  call       ""Function System.Runtime.CompilerServices.TaskAwaiter(Of Integer).GetResult() As Integer""
+    IL_006b:  ldloca.s   V_1
+    IL_006d:  initobj    ""System.Runtime.CompilerServices.TaskAwaiter(Of Integer)""
+    IL_0073:  stfld      ""B.x As Integer""
+    IL_0078:  leave.s    IL_009c
+  }
+  catch System.Exception
+  {
+    IL_007a:  dup
+    IL_007b:  call       ""Sub Microsoft.VisualBasic.CompilerServices.ProjectData.SetProjectError(System.Exception)""
+    IL_0080:  stloc.2
+    IL_0081:  ldarg.0
+    IL_0082:  ldc.i4.s   -2
+    IL_0084:  stfld      ""Program.VB$StateMachine_0_Assign.$State As Integer""
+    IL_0089:  ldarg.0
+    IL_008a:  ldflda     ""Program.VB$StateMachine_0_Assign.$Builder As System.Runtime.CompilerServices.AsyncTaskMethodBuilder""
+    IL_008f:  ldloc.2
+    IL_0090:  call       ""Sub System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetException(System.Exception)""
+    IL_0095:  call       ""Sub Microsoft.VisualBasic.CompilerServices.ProjectData.ClearProjectError()""
+    IL_009a:  leave.s    IL_00b1
+  }
+  IL_009c:  ldarg.0
+  IL_009d:  ldc.i4.s   -2
+  IL_009f:  dup
+  IL_00a0:  stloc.0
+  IL_00a1:  stfld      ""Program.VB$StateMachine_0_Assign.$State As Integer""
+  IL_00a6:  ldarg.0
+  IL_00a7:  ldflda     ""Program.VB$StateMachine_0_Assign.$Builder As System.Runtime.CompilerServices.AsyncTaskMethodBuilder""
+  IL_00ac:  call       ""Sub System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetResult()""
+  IL_00b1:  ret
+}")
+        End Sub
+
+        <Fact>
+        <WorkItem(42755, "https://github.com/dotnet/roslyn/issues/42755")>
+        Public Sub KeepLtrSemantics_CallOnClassFieldOnClass()
```

Comment:
I haven't actually changed anything on Calls, since previously we would always check side effects, and now we explicitly pass `shouldCheckSideEffects:=True` for all cases other than assignments.

However I added a  couple of basic tests as sanity checks to confirm this.